confidence/models.py

```diff
@@ -233,36 +229,17 @@ def __setstate__(self, state: typing.Dict[str, typing.Any]) -> None:
 
         if isinstance(self._missing, Missing):
             # reverse the Missing encoding done in __getstate__
-            self._missing = _MISSING_MAPPING[self._missing]
-
-
-class _NotConfigured(Configuration):
-    """
-    Sentinel value to signal there is no value for a requested key.
-    """
-    def __bool__(self) -> bool:
-        return False
-
-    def __repr__(self) -> str:
-        return '(not configured)'
-
-    __str__ = __repr__
+            self._missing = {Missing.silent: NotConfigured,
+                             Missing.error: _NoDefault}[self._missing]
 
 
-# TODO: can't reference NotConfigured yet here, but instantiation needs the value, maybe a cleaner solution is needed?
-_MISSING_MAPPING: typing.MutableMapping[Missing, typing.Any] = {Missing.silent: None,
-                                                                Missing.error: NoDefault}
-
-
-# FIXME: ordering of things in this file is a mess (some of it because of reasons, needs cleanup)
-
-
-# overwrite NotConfigured as an instance of itself, a Configuration instance without any values
-NotConfigured = _NotConfigured()
-# NB: NotConfigured._missing refers to the NotConfigured *class* at this point, fix this after the name override
+NotConfigured = type("NotConfigured", (Configuration,), {
+    "__bool__": lambda self: False,
+    "__repr__": lambda self: '(not configured)',
+    "__str__": lambda self: '(not configured)',
+})
+NotConfigured = NotConfigured()
```

Comment:
Ghe, the ol' 3-arg `type` call :D Does mypy allow this line though?

Reply:
Interestingly, it does, maybe `mypy` handles class defs differently from types defined like this. Might be considered a 'bug' in mypy for now, but if it works... :man_shrugging:

Reply:
Yeah it seems to assign the type "type" to any object that was created through this kind of type definition. So maybe mypy inherently treats classes created through class defs differently from other objects? 

If you run mypy with `allow_redefinition = False` it will still raise an error (albeit not at this particular line), so mypy is still not 100% okay with how `NotConfigured` is treated throughout the rest of the code. But it just so happens to be an error we allow :D